v2/tools/generator/internal/astmodel/package_import_set.go

```diff
@@ -184,23 +186,62 @@ func (set *PackageImportSet) orderImports(i PackageImport, j PackageImport) bool
 	return i.packageReference.ImportPath() < j.packageReference.ImportPath()
 }
 
-func (set *PackageImportSet) createMapByGroup() map[string][]PackageImport {
-	result := make(map[string][]PackageImport)
+// tryAssignImportAliases attempts to apply the specified PackageImportStyle to the specified imports.
+// Returns true if all the imports had unique names and were successfully updated, false otherwise.
+// Does not modify the set if it returns false.
+func (set *PackageImportSet) tryAssignImportAliases(
+	style PackageImportStyle,
+) bool {
+	// aliased keeps track of every import that we've assigned an alias to, keyed by the new alias.
+	// This allows us to check for duplicates.
+	aliased := make(map[string]PackageImport, len(set.imports))
+
+	// groupConflicts keeps track of any group for which we have conflicted aliases.
+	// This allows us to add aliases for groups where we don't have any conflicts.
+	groupConflicts := make(map[string]bool, 100)
```

Comment:
minor: Can you use our `set.Set[T]` here rather than a map to bool?

Reply:
Done - with a workaround to avoid the receivers on this type also being `set`.